### ISSUE TITLE
v1 - TypeError: Cannot set property '__resolveType' of undefined when resolving union type

### ISSUE THREAD

**Issue by zicodeng** [Urigo/graphql-modules#1313]
Hi, I am getting this error for resolving `union` type:

```
TypeError: Cannot set property '__resolveType' of undefined
```

Repro code looks like this:

```
const { ApolloServer } = require('apollo-server');
const { createModule, createApplication, gql } = require('graphql-modules');

const DATA = {
  id: 'data',
  union: {
    title: 'Jack',
  },
};

const myModule = createModule({
  id: 'my-module',
  dirname: __dirname,
  typeDefs: gql`
    type Post {
      title: String
    }

    type Book {
      author: String
    }

    union MyUnion = Post | Book

    type Data {
      id: String
      union: MyUnion
    }

    type Query {
      myData: Data
    }
  `,
  resolvers: {
    MyUnion: {
      __resolveType: (parent) => {
        if (parent.title) return 'Post';
        else return 'Book';
      },
    },
    Query: {
      myData: () => {
        return DATA;
      },
    },
  },
});

const application = createApplication({
  modules: [myModule],
});

const schema = application.createSchemaForApollo();

const server = new ApolloServer({
  schema,
});

server.listen().then(({ url }) => {
  console.log(`🚀 Server ready at ${url}`);
});
```

Did I miss anything? At lease this is what I did for v0 and had no problem. It only occurs on v1 🤔

**Comment by zicodeng** [comment]
@ardatan any update on this one? Is it acknowledged by the team? thanks!

**Comment by zicodeng** [comment]
This seems to be a very critical bug which makes union unusable. Can we get some attention? Thanks!

**Comment by kamilkisiela** [comment]
@zicodeng we fixed it in #1344
